### PR TITLE
Modernize build system

### PR DESCRIPTION
**Pull request by AlbertDeFusco**
### Description

Here I'm updating the output Python package to 
* use `src/<package_name>` dir structure
* use pyproject.toml
    * with hatch and hatch-vcs instead of versioneer
* added pre-commit with [ruff](https://github.com/astral-sh/ruff) linting

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: {{cookiecutter.repo_name}}/conda.recipe/meta.yaml [hunk 1]
```diff
@@ -19,19 +20,15 @@ build:
   # "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or
   # "skip: True  # [not win]" to limit to Windows.
   {% raw -%}
-  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed -vv .
+  script: SETUPTOOLS_SCM_PRETEND_VERSION={{version}} {{ PYTHON }} -m pip install --no-build-isolation --no-deps --ignore-installed -vv .
```
Comment (kenodegard) [inline {{cookiecutter.repo_name}}/conda.recipe/meta.yaml]: Alternatively adding `git_archival.txt` removes the need for `SETUPTOOLS_SCM_PRETEND_VERSION`: https://setuptools-scm.readthedocs.io/en/stable/usage/#git-archives

**Review by kenodegard — COMMENTED** [review]
Perhaps it would be good to also add a GH Actions workflow template?

This also looks pretty similar to the [conda plugin cookiecutter template](https://github.com/conda/conda-plugin-template/tree/main/cookiecutter-template), should we work to combine them? Maybe move the conda plugin cookiecutter template to this repo and rename conda-plugin-template to conda-plugin-tutorials?

cc @travishathaway @beeankha

File: {{cookiecutter.repo_name}}/conda.recipe/meta.yaml (hunk 1 above)
Comment (AlbertDeFusco) [reply {{cookiecutter.repo_name}}/conda.recipe/meta.yaml]: that did it, thanks!

**Comment by travishathaway** [comment]
@kenodegard,

I think it's best to keep the scope of this cookie cutter repository narrow and not include other templates. The plugins template is different enough that it doesn't make sense to combine them into a single one.

I would like to thank @AlbertDeFusco for this pull request and showing this repository a little TLC 👏. We should keep improving it. For example, it would be cool to include options to use to the new recipe format with `rattler-build`.

**Review by travishathaway — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,53 +1,66 @@
...
+    [5/9] package_name (test_pkg): (optional, autogenerated from repository-name)
+    [6/9] project_short_description (Short description): (optional)
+    [7/9] noarch_python (y): (optional, default yes)
+    [8/9] include_cli (y): (optional, default yes)
+    [9/9] Select open_source_license
+      1 - MIT
+      2 - BSD
+      3 - ISC
+      4 - Apache
+      5 - GNUv3
```
Comment (jezdez) [inline README.md:42]: Per conda governance policy non-permissive licenses are highly discouraged

**Review by jezdez — APPROVED** [review]

**Pull request merged**